modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/repo/impl/DetectorRepositoryImpl.java

```diff
@@ -104,6 +103,8 @@ public void updateDetector(String uuid, Detector detector) {
                 if (name.equals("lastUpdateTimestamp")) {
```

Comment:
Please reverse the order to “lastUpdateTimestamp”.equals(name) (similarly below) to avoid possible NPEs. This is a general habit I recommend, not limited to this PR.